.pre-commit-config.yaml

```diff
@@ -0,0 +1,45 @@
+ci:
+  autoupdate_schedule: "quarterly"
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+  - id: check-added-large-files
+  - id: check-case-conflict
+  - id: check-executables-have-shebangs
+  - id: check-symlinks
+  - id: check-toml
+  - id: check-yaml
+  - id: debug-statements
+  - id: destroyed-symlinks
+  - id: end-of-file-fixer
+    exclude: '\.(graphml)$'
+  - id: trailing-whitespace
+    exclude: '\.(pdb|gro|top|sdf|xml|cif|graphml)$'
+- repo: https://github.com/psf/black
+  rev: 24.2.0
```

Comment:
I know it's possible with black to choose a "standard", is this the way we do this or does this just control the version of black we install?

Reply:
I'm also a bit curious here. Which tool is responsible for installing what? For instance, with black, does running the pre-commit install actually install black or just the hook that will execute black?

Reply:
Good question! So if you are just using pre-commit.ci, locally you don't need to do anything. But if you use this locally, `pre-commit` when running for the first time will create a virtual python environment (separate than whatever one you are currently using, this lets you use packages and versions that might not work with your program be used for the pre-commit hooks) and install everything needed for the hooks and run them. When you re-run the hooks, it will re-use the virtual environment it created the first time.

so tl;dr if you install pre-commit, it will create a virtual env that will install all the tools needed, then execute the hooks

Reply:
Also as a quick follow up, there is a section in the pyproject.toml:

```
[tool.black]
line-length = 120 
```

that controls black's config, so we can set it to use whichever "standard" we want.